WebApiTest/AudioControllerTests.cs

```diff
@@ -0,0 +1,158 @@
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Http;
+using System.Collections.Generic;
+using SoftServe.ITAcademy.BackendDubbingProject.Administration.Core.DTOs;
+using System.IO;
+using Microsoft.AspNetCore.Http.Internal;
+using System.Text;
+using System.Web;
+
+namespace SoftServe.ITAcademy.BackendDubbingProject.WebApiTest
+{
+    [ExcludeFromCodeCoverage]
+    public class AudioControllerTests
+    {
+        [Fact]
+        public async Task TestGetAll_200OK()
+        {
+            // Arrange
+            var mock = new Mock<IAdministrationService>();
+            mock.Setup(service => service.GetAllAudiosAsync()).Returns(async () => { return DullData.GetAllAudioDTOs(); });
+            var controller = new AudioController(mock.Object);
+
+            // Act
+            var result = (await controller.GetAll()).Result as ObjectResult;
+            var value = result.Value as List<AudioDTO>;
+
+            // Assert
+            Assert.NotEmpty(value);
+            Assert.Equal(5, value.Count);
+            Assert.Equal(StatusCodes.Status200OK, result.StatusCode);
+        }
+
+        [Fact]
+        public async Task TestGetById_200OK()
+        {
+            // Arrange
+            var mock = new Mock<IAdministrationService>();
+            int id = 1;
+            mock.Setup(service => service.GetAudioByIdAsync(id)).Returns(async () => {
+                return new AudioDTO { Id = 1, FileName = "File1", OriginalText = "Text1", SpeechId = 1, LanguageId = 1 };
+            });
+            var controller = new AudioController(mock.Object);
+
+            // Act
+            var result = (await controller.GetById(id)).Result as ObjectResult;
+            var value = result.Value as AudioDTO;
+
+            // Assert
+            Assert.NotNull(value);
+            Assert.Equal(id, value.Id);
+            Assert.Equal(StatusCodes.Status200OK, result.StatusCode);
+        }
+
+        [Fact]
+        public async Task TestGetById_404NotFound()
+        {
+            // Arrange
+            var mock = new Mock<IAdministrationService>();
+            int id = 1;
+            mock.Setup(service => service.GetAudioByIdAsync(id)).Returns(async () => { return null; });
+            var controller = new AudioController(mock.Object);
+
+            // Act
+            var result = (await controller.GetById(id)).Result as IStatusCodeActionResult;
+
+            // Assert
+            Assert.Equal(StatusCodes.Status404NotFound, result.StatusCode);
+        }
+
+        [Fact]
+        public async Task TestCreate_201Created()
+        {
+            // Arrange
+            var mock = new Mock<IAdministrationService>();
+            var controller = new AudioController(mock.Object);
+            AudioDTO audio = new AudioDTO { Id = 1, FileName = "File1", OriginalText = "Text1", SpeechId = 1, LanguageId = 1 };
```

Comment:
move values to constants like SomeFileName, OtherFileName
SomeSpeechId etc. it will help reuse test setups